src/categories/views/CategoryList/CategoryList.tsx

```diff
@@ -143,7 +143,7 @@ export const CategoryList: React.FC<CategoryListProps> = ({ params }) => {
   return (
     <>
       <CategoryListPage
-        categories={mapEdgesToItems(data?.categories)}
+        categories={mapEdgesToItems(data?.categories) || []}
```

Comment:
If it was the same component over and over again included and we'd pass `data || []` as props then it would make sense to keep code DRY and less explicit, but in this case it would be making data flow less obvious because we'd have to add default prop to all lists. And I'm pretty sure we'd have to refactor it once again after we set `strictNullCheck: true` to `tsconfig` because ts would be demanding passing a non-undefined value here anyway (unless we'd type it as `data | undefined` which wouldn't be semantically correct).
I'd go with your proposal if it was a generic component, but it's not, so it seems to me like moving code from an expected place to less obvious one and not gaining anything in return (because you need to do it for each component anyway).

Reply:
I can't agree. Imo the most obvious place to keep default props for given component is inside of that component, not outside of it. After all, in the component is the place that actually handles what to display in given scenarios.  Especially, since, as you said, generic components should have this set as default prop (that makes 2 different ways to handle it already) and there are cases in the app e.g `components/ShippingZonePostalCodes/ShippingZonePostalCodes.tsx` where there is a list of `postalCodes` passed as prop that is handled differently depending on whether its `[]` or `undefined`. So we're using `Type | undefined` and padding `postalCodes` without `|| []` which makes 3 different ways. We also discussed this typing previously [here](https://github.com/saleor/saleor-cloud-billing/pull/258#discussion_r627311638). I think the approach I proposing is simply the most consistent one, so adding default props is not related to logic or implementation of given component and doesn't require to be typed differently once the component changes its status from generic to non generic or the other way around or when we actaully decide we would like to add logic to it when the props is `undefined`.

Reply:
We're doing it exactly as we established in the discussion you provided. Components that `need` the list are provided with `[]` explicitly, handling it in a component would make these props optional therefore semantically incorrect. I believe I wasn't clear about the generic component part - it would make sense to set it as default only if all those components I fixed were the same one, because we'd be practical rather than blindly follow convention (an exception here would be acceptable I think), but it's not the case and will never be.
Also it's not the place to make that choice here because even if we'd go for default prop, it should be done in a project-wide refactor to keep code consistent.